db/kv_dense_block.go

```diff
@@ -41,7 +41,12 @@ type DenseBlock struct {
 	Key        string              // Key of block document in the index bucket
 	value      []byte              // Binary storage of block data, in the above format
 	cas        uint64              // Document cas
-	clock      base.PartitionClock // Highest seq per vbucket written to the block
+
+	// Highest seq per vbucket written to the block.
+	// Lazily-loaded because DenseBlock is used for both readers and writers, and readers don't care about the cumulative clock.
+	// Should never be read directly, instead call getClock(), otherwise you might hit issues like SG #3026.
+	_clock      base.PartitionClock
```

Comment:
Just a style question - I feel like this kind of mid-struct comment makes it harder to see at a glance the set of properties in a struct.  I'm fine with including verbose comments for property definition and usage - what do you think about including these in a struct-level comment prior to the struct definition?

Reply:
I trimmed it down and moved part of the comment to the struct defn